packages/appium/lib/main.js

```diff
@@ -276,6 +278,9 @@ async function init(args) {
       pluginConfig,
       appiumHome,
     });
+  } else if (isSetupCommandArgs(preConfigArgs)) {
+    await setupCommand(appiumHome, preConfigArgs, driverConfig, pluginConfig);
+    return /** @type {InitResult<Cmd>} */ ({});
```

Comment:
maybe the return value should be the merge of the return values of all the install commands, in case this is run by script?

Reply:
Meaning by `appium setup`? The current behavior is if the command succeeds, the existing status will be `0`. Otherwise non-zero,  so probably users can catch the error if an error occurred.

current e.g.
```
% ./node_modules/.bin/appium setup
dbug Appium No manifest file found at /Users/kazu/.appium/node_modules/.cache/appium/extensions.yaml; creating
dbug Appium Discovering newly installed extensions...
✔ Checking if 'appium-uiautomator2-driver' is compatible
✔ Installing 'uiautomator2' using NPM install spec 'appium-uiautomator2-driver'
ℹ Driver uiautomator2@3.5.1 successfully installed
- automationName: UiAutomator2
- platformNames: ["Android"]
✔ Checking if 'appium-xcuitest-driver' is compatible
✔ Installing 'xcuitest' using NPM install spec 'appium-xcuitest-driver'
ℹ Driver xcuitest@7.15.3 successfully installed
- automationName: XCUITest
- platformNames: ["iOS","tvOS"]
✔ Checking if '@appium/images-plugin' is compatible
✔ Installing 'images' using NPM install spec '@appium/images-plugin'
ℹ Plugin images@3.0.6 successfully installed
% echo $?                         
0
```

Reply:
i meant, in `--json` mode it can return a nice json object which is the merge of all the install operations.

Reply:
-> https://github.com/appium/appium/issues/20130 I'll check and update the `--json` mode